pkg/configmap/configmap.go

```diff
@@ -0,0 +1,187 @@
+	"sort"
+	"time"
+)
+
+type (
+	Service interface {
+		ListNamesAndLabels() (configMapNames, labels []string, err error)
+		List(listOptions metav1.ListOptions) (configMaps []v1.ConfigMap, err error)
+		GetUnused(namespace string, configMaps []v1.ConfigMap) (unusedConfigMaps []v1.ConfigMap, funcErr error)
+		Delete(configMaps []v1.ConfigMap)
+		FilterByTime(configMaps []v1.ConfigMap, olderThan time.Time) (filteredConfigMaps []v1.ConfigMap)
+		FilterByMaxCount(configMaps []v1.ConfigMap, keep int) (filteredConfigMaps []v1.ConfigMap)
+		Print(configMaps []v1.ConfigMap)
+	}
+	ConfigMapsService struct {
+		configuration Configuration
+		client        core.ConfigMapInterface
+		helper        kubernetes.Kubernetes
+	}
+)
+
+type Configuration struct {
+	Batch bool
+}
+
+// NewConfigMapsService creates a new Service instance
+func NewConfigMapsService(client core.ConfigMapInterface, helper kubernetes.Kubernetes, configuration Configuration) ConfigMapsService {
+	return ConfigMapsService{
+		client:        client,
+		helper:        helper,
+		configuration: configuration,
+	}
+}
+
+// ListNamesAndLabels return names and labels of Config Maps
+func (cms ConfigMapsService) ListNamesAndLabels() (resourceNames, labels []string, err error) {
+	configMaps, err := cms.List(metav1.ListOptions{})
+	var objectMetas []metav1.ObjectMeta
+	for _, cm := range configMaps {
+		objectMetas = append(objectMetas, cm.ObjectMeta)
+	}
+	configMapNames, labels := util.GetNamesAndLabels(objectMetas)
+	return configMapNames, labels, nil
+}
+
+// List returns a list of ConfigMaps from a namespace
+func (cms ConfigMapsService) List(listOptions metav1.ListOptions) ([]v1.ConfigMap, error) {
+	configMaps, err := cms.client.List(listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	return configMaps.Items, nil
+}
+
+// GetUnused return unused Config Maps
+func (cms ConfigMapsService) GetUnused(namespace string, configMaps []v1.ConfigMap) (unusedConfigMaps []v1.ConfigMap, funcErr error) {
+	var usedConfigMaps []v1.ConfigMap
+	funk.ForEach(openshift.PredefinedResources, func(predefinedResource schema.GroupVersionResource) {
+		funk.ForEach(configMaps, func(resource v1.ConfigMap) {
+
+			resourceName := resource.GetName()
+
+			if funk.Contains(usedConfigMaps, resource) {
+				// already marked as existing, skip this
+				return
+			}
+			contains, err := cms.helper.ResourceContains(namespace, resourceName, predefinedResource)
+			if err != nil {
+				funcErr = err
+				return
+			}
+
+			if contains {
+				usedConfigMaps = append(usedConfigMaps, resource)
+			}
+		})
+	})
+
+	for _, resource := range configMaps {
+		if !funk.Contains(usedConfigMaps, resource) {
+			unusedConfigMaps = append(unusedConfigMaps, resource)
+		}
+	}
+
+	return unusedConfigMaps, funcErr
+}
+
+// Delete removes Config Maps
+func (cms ConfigMapsService) Delete(configMaps []v1.ConfigMap) {
+	for _, resource := range configMaps {
+		namespace := resource.GetNamespace()
```

Comment:
In Go, for known data structs you can directly access the public properties, no need to call `GetNamespace`
If you use `resource.Namespace` (and the other properties) in the 2 occurrences below, you save yourselves a bunch of redundant variables and code lines, making it more readable.